app/views/layouts/application.html.slim

```diff
@@ -9,7 +9,10 @@ html
     = stylesheet_pack_tag 'application', media: 'all', 'data-turbolinks-track': 'reload'
     = javascript_pack_tag 'application', 'data-turbolinks-track': 'reload'
   body.wrapper
+  div.filter_block
     = render 'layouts/flash_messages'
     = render 'layouts/navigation'
+    - if controller_name != 'sessions' && controller_name != 'registrations' && controller_name != 'passwords'
```

Comment:
Pls try to simplify this logic. In order to follow DRY rule you can move this line to controller (for example).
`%w(sessions registrations passwords).include?(controller_name)` or even find another solution how to deal with views restrictions.